subprojects/gradle/kotlin-dsl-support/src/test/kotlin/com/avito/kotlin/dsl/OptionalFilePropertyTest.kt

```diff
@@ -0,0 +1,181 @@
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.io.TempDir
+import java.io.File
+import javax.inject.Inject
+
+internal class OptionalFilePropertyTest {
+
+    abstract class OptimisticTestTask : DefaultTask() {
+
+        @get:InputFile
+        @get:Optional
+        abstract val someOptionalFile: RegularFileProperty
+
+        @TaskAction
+        fun doStuff() {
+            val file = someOptionalFile.orNull?.asFile
+            assertThat(file?.readText()).isEqualTo("bla bla bla")
+        }
+    }
+
+    abstract class PessimisticTestTask : DefaultTask() {
+
+        @get:InputFile
+        @get:Optional
+        abstract val someOptionalFile: RegularFileProperty
+
+        @TaskAction
+        fun doStuff() {
+            val file = someOptionalFile.orNull?.asFile
+            assertThat(file).isNull()
+        }
+    }
+
+    abstract class PessimisticWorkerTestTask @Inject constructor(
+        private val workerExecutor: WorkerExecutor
+    ) : DefaultTask() {
+
+        @get:InputFile
+        @get:Optional
+        abstract val someOptionalFile: RegularFileProperty
+
+        @TaskAction
+        fun doStuff() {
+            workerExecutor.submit(SomeWorkerAction::class.java) { workerConfiguration: WorkerConfiguration ->
+                workerConfiguration.isolationMode = IsolationMode.NONE
+                workerConfiguration.setParams(someOptionalFile.orNull?.asFile)
+            }
+        }
+    }
+
+    class SomeWorkerAction @Inject constructor(private val file: File?) : Runnable {
+        override fun run() {
+            assertThat(file).isNull()
+        }
+    }
+
+    @Test
+    fun `optional file - passed to task - when non-empty file set`(@TempDir tempDir: File) {
+
+        val project = ProjectBuilder.builder().build()
+
+        val file = File(tempDir, "x.txt").apply {
+            createNewFile()
+            writeText("bla bla bla")
+        }
+
+        val task = project.tasks.register<OptimisticTestTask>("testTask") {
+            someOptionalFile.set(project.optionalIfNotExists(file))
+        }
+
+        task.get().doStuff()
+    }
+
+    @Test
+    fun `optional file - resolved to null in task - when nothing is set`() {
+
+        val project = ProjectBuilder.builder().build()
+
+        val task = project.tasks.register<PessimisticTestTask>("testTask") {}
+
+        task.get().doStuff()
+    }
+
+    @Test
+    fun `optional file - resolved to null in task - when empty file is set`() {
+
+        val project = ProjectBuilder.builder().build()
+
+        val nonExistentFile = File("x.txt")
+
+        val task = project.tasks.register<PessimisticTestTask>("testTask") {
+            someOptionalFile.set(project.optionalIfNotExists(nonExistentFile))
+        }
+
+        task.get().doStuff()
+    }
+
+    @Test
+    fun `optional file - resolved to null in task - when empty file is set using worker api`(@TempDir projectDir: File) {
+
+        projectDir.file(
```

Comment:
Why do we need that test? What problems with worker api did we have?

Reply:
i already deleted it
trying right now to find out why we fail in same scenario on real project